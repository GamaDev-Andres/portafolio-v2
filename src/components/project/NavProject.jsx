import { Link } from 'react-router-dom'

const NavProject = () => {
  return (
    <nav className="fixed w-full  sm:top-0 sm:left-0 z-10">
      <div className="flex relative justify-center max-w-[1256px] mx-auto bg-base items-center px-4 py-2">
        <Link to="/">
          <h1 className="text-letra font-black text-4xl px-4 text-center">
            GamaDev
          </h1>
        </Link>
        <Link className="absolute top-4 right-4" to="/">
          <i className="sm:mr-2 fa-solid fa-house"></i>
        </Link>
      </div>
    </nav>
  )
}

export default NavProject
