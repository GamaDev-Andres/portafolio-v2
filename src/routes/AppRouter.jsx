import { useState } from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import Home from '../components/home/Home';
import Project from '../components/project/Project';

function AppRouter() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/project/:id" element={<Project />} />
      </Routes>
    </BrowserRouter>
  );
}

export default AppRouter;
