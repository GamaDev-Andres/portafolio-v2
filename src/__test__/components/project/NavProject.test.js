/**
 * @jest-environment jsdom
 */
import Enzyme from 'enzyme'
import { HelmetProvider } from 'react-helmet-async'
import { MemoryRouter } from 'react-router-dom'
import NavProject from '../../../components/project/NavProject'
import AppRouter from '../../../routes/AppRouter'

describe('probando <NavProject/>', () => {
  test('debe montarse correctamente', () => {
    const wrapper = Enzyme.mount(
      <MemoryRouter>
        <NavProject />
      </MemoryRouter>)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('h1').text().trim()).toBe('GamaDev')
  })
  test('debe montarse correctamente en ruta "project" ', () => {
    const wrapper = Enzyme.mount(
      <MemoryRouter initialEntries={['/project/BARBERAPP']}>
        <HelmetProvider>
          <AppRouter />
        </HelmetProvider>
      </MemoryRouter>
    )
    expect(wrapper.find('NavProject').exists()).toBe(true)
  })
})
