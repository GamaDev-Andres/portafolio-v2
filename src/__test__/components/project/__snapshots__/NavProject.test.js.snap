// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probando <NavProject/> debe montarse correctamente 1`] = `
<nav
  className="fixed w-full  sm:top-0 sm:left-0 z-10"
>
  <div
    className="flex relative justify-center max-w-[1256px] mx-auto bg-base dark:bg-baseDark items-center px-4 py-2"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h1
        className="text-letra font-black text-4xl px-4 text-center dark:text-letraDark"
      >
        GamaDev
      </h1>
    </a>
    <a
      className="absolute top-4 right-4"
      href="/"
      onClick={[Function]}
    >
      <i
        className="sm:mr-2 fa-solid fa-house"
      />
    </a>
  </div>
</nav>
`;
